### DISCUSSION TITLE
How to create models that work consistently

### DISCUSSION THREAD

**Discussion by hadronknights** [FIRST-Tech-Challenge/fmltc#290] [Q&A]
We have been using ftc ml tool chain to develop models to detect team signals this season. We have tried multiple types of signals (objects) but we are unable to detect the signal consistently (maybe once in 10 attempts) during autonomous. Any recommendation how we can make the model more reliable. Is light an issue? If so, teams will never have lighting condition similar to tournament, therefore model created in a team practice area from school libraries to garages has to work in the tournament fields. Looking for expert guidance here on how to improve the object detection.

FYI: FTC provided model Powerplay.tflite works consistently though.

**Comment by texasdiaz** [comment]
I'll take this as a great compliment, thanks - the model working great with different lighting conditions took a lot of work to understand what was going on, and a couple iterations of trying various techniques. 

Have you done a search on the FTC community? Probably the best discussion on this topic can be found here:
https://ftc-community.firstinspires.org/t/proper-process-for-object-detection/277

Once you've read that post and followed the link and read the linked document in that post, if you have further questions please create a topic post in the FTC Community forum and I'll do my best to help you out.
